tree/tree/src/TTree.cxx

```diff
@@ -1224,8 +1223,8 @@ bool CheckReshuffling(TTree &mainTree, TTree &friendTree)
          return false;
       auto *majorFormula = idx->GetMajorFormulaParent(&mainTree);
       auto *minorFormula = idx->GetMinorFormulaParent(&mainTree);
-      if ((majorFormula == nullptr || majorFormula->GetNdim() == 0) ||
-          (minorFormula == nullptr || minorFormula->GetNdim() == 0))
+      if ((majorFormula == nullptr || idx->GetFormulaNdim(majorFormula) == 0) ||
```

Comment:
Maybe just
```
if (idx->IsValidFor(mainTree))
```
i.e. adding 1 function instead of 2 and rolling all the code here into it.  Thanks.

Reply:
@pcanal Can you check now? I hope I understand that you want